### ISSUE TITLE
WebDriver.IO and ChromeDriver Infinite Loop Fetching an element that does not exist

### ISSUE THREAD

**Issue by renatofmachado** [webdriverio/webdriverio#5290]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 6.0.16
 - **Mode:** Test Runner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** v12.13.1 (also happens on 10.16)
 - **NPM version:** 6.12.1
 - **Browser name and version:** Chrome 81
 - **Platform name and version:** MacOS & Docker
 - **Additional wdio packages used (if applicable):**
    - [x] "@wdio/cli": "^6.0.16",
    - [x] "@wdio/local-runner": "^6.0.16",
    - [x] "@wdio/mocha-framework": "^6.0.16",
    - [x] "@wdio/spec-reporter": "^6.0.16",
    - [x] "@wdio/sync": "^6.0.16"
    - [x] "wdio-chromedriver-service": "^6.0.2",
    - [x] "chromedriver": "81.0.0",

**Config of WebdriverIO**
Pretty much the standard configuration, just changed the max instances to 6 and I'm using Mocha.

**Describe the bug**
In my project, I have a list of elements. An element can be "active" or "inactive". If the element is considered "active", I'll show a chip saying that element is active. If the element is considered "inactive", I'll show a button to activate the element.

So, I have a page object that does the following:

```
class Elements {
  get module() {
    return $('[data-id="elements-list"]');
  }

  get listOfElements() {
    return this.module.$$('.card');
  }

  get notConfiguredElements() {
    return this.listOfElements.filter(card => card.$('[data-id="add-element-button"]').isExisting());
  }

  get alreadyConfiguredElements() {
     return this.listOfElements.filter(card => card.$('[data-id="already-added-chip"]').isExisting());
  }
}
```

So, as you can see, since I go through the list of elements (they can either be active or inactive), which makes the `card.$(selector).isExisting()` not find that selector in some cases.

**To Reproduce**

I've created a fresh project, using all the defaults of Webdriver.io (following this [tutorial](https://webdriver.io/docs/gettingstarted.html)).

In here, I've set the following test that reproduces the behavior stated above:

```
it('accesses unreachable element', () => {
        browser.url('https://webdriver.io')
        const text = $('asdasdasdas').$('asdkljasldjwlqkejqwlke').getText()

        assert.strictEqual(text, 'hello')
    })
```

If I try to reach any unreachable element with the following selector: `$('asdasd')` it works fine, and returns that the selector does not exist. If I do `$('asasdasd').$('asdasdas')` or `$('asdasd').getText()` it just enters an infinite loop that tries to fetch that element.

**Expected behavior**

In my project, when I try to filter by the elements, I expect to get the correct array of elements (not include the elements that are not found).
For the demo I built using the default values, I would expect to return an error saying that the `asdasdasdas` selector was not found.

**Log**

https://gist.github.com/renatofmachado/fff703a420bfe4314ae1f2ad5ca3f42d

**Additional context**

Back in December all of that code just worked. Since then, I've upgraded from Chrome 79 to 80 and 81. It failed ever since. Can someone give me some hints on what I may be doing wrong? 🙏

**Comment by tawfiknouri** [comment]
@renatofmachado  i can't reproduce this bug

[chrome  windows #0-0] webdriver.io page
[chrome  windows #0-0]    ✖ accesses unreachable element
[chrome  windows #0-0]
[chrome  windows #0-0] 1 failing (10.2s)
[chrome  windows #0-0]
[chrome  windows #0-0] 1) webdriver.io page accesses unreachable element
[chrome  windows #0-0] Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\Users\tawfik.nouri\Downloads\webdriverio-react-typescript-master\test\specs\basic.ts)
[chrome  windows #0-0] Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\Users\tawfik.nouri\Downloads\webdriverio-react-typescript-master\test\specs\basic.ts)
[chrome  windows #0-0]     at Test.Runnable._timeoutError (C:\Users\tawfik.nouri\Downloads\webdriverio-react-typescript-master\node_modules\mocha\lib\runnable.js:463:10)
[chrome  windows #0-0]     at Timeout.<anonymous> (C:\Users\tawfik.nouri\Downloads\webdriverio-react-typescript-master\node_modules\mocha\lib\runnable.js:267:24)
[chrome  windows #0-0]     at listOnTimeout (internal/timers.js:531:17)
[chrome  windows #0-0]     at processTimers (internal/timers.js:475:7)


Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:16

Could you share a simple github project to reproduce this bug ?

**Comment by renatofmachado** [comment]
I've recreated the issue here: https://github.com/renatofmachado/wdio-issue

Just run `yarn` and then `yarn acceptance` 👍

**Comment by tawfiknouri** [comment]
@renatofmachado 
It seems the infinite loop it created by your code 

Try this code, works fine for me
```
import { expect } from 'chai';

describe('webdriver.io page', () => {
	it('should have the right title', () => {
		browser.url('http://localhost:3000/');

		var listOfCards = $('[data-id="elements-list"]').$$('.card');
		var alreadyConfiguredElements = listOfCards.filter(card => card.$('[data-id="already-added-chip"]').isExisting());
		console.log("alreadyConfiguredElements ==> " + alreadyConfiguredElements.length);

		var notConfiguredElements = listOfCards.filter(card => card.$('[data-id="add-element-button"]').isExisting());
		console.log("notConfiguredElements ==> " + notConfiguredElements.length);

		expect(alreadyConfiguredElements.length).equal(18);
		expect(notConfiguredElements.length).equal(6);

	});
});
```

**Comment by renatofmachado** [comment]
Just tried it, and it didn't work for me. 

I've created a Dockerfile & docker-compose. And it fails there as well. Do a git pull and execute it using the following command: `docker-compose down; docker-compose build acceptance-tests; docker-compose run --rm acceptance-tests; docker-compose down;`

It just runs until it gets a time out. Never manages to complete the test.

Thanks for the help so far! 🙏

**Comment by tawfiknouri** [comment]
@renatofmachado could you try with **services: ['selenium-standalone']** (wdio.conf.js)
You need to install selenium-standalone-service `npm i @wdio/selenium-standalone-service --save-dev`

but yes it seems a bug, (my script work file with selenium-standalone) ...

**Comment by renatofmachado** [comment]
I've tested with selenium and it works 👍

Maybe it's a bug with the ChromeDriver itself or maybe something got changed?

Also, if I just use direct queries using only the ChromeDriver it works as well.. it just goes into an infinite loop when iterating on all of those elements and doing sub queries on them. But it just happens when we have a lot of elements (e.g. it didn't happen when I had only 4 cards).

Thank you for your help 🙏
